app/ts/utils/requests.ts

```diff
@@ -68,12 +68,46 @@ export async function fetchWithTimeout(resource: RequestInfo | URL, init?: Reque
 }
 
 export const safeGetTab = async (tabId: number) => {
-    const tab = await browser.tabs.get(tabId)
-    try {
-		const error = browser.runtime.lastError
-		if (error !== undefined && error.message !== undefined) throw new Error(error.message)
-        return tab
-    } catch (e: unknown){
+	const tab = await browser.tabs.get(tabId)
+	try {
+		checkAndThrowRuntimeLastError()
+		return tab
+	} catch (e: unknown){
+		return undefined
+	}
+}
+
+export const safeGetWindow = async (windowId: number) => {
+	const tab = await browser.windows.get(windowId)
+	try {
+		checkAndThrowRuntimeLastError()
+		return tab
+	} catch (e: unknown){
 		return undefined
-    }
+	}
+}
+
+export const updateTabIfExists = async (tabId: number, updateProperties: browser.tabs._UpdateUpdateProperties) => {
+	try {
+		const tab = await browser.tabs.update(tabId, updateProperties)
+		checkAndThrowRuntimeLastError()
+		return tab
+	} catch (e: unknown){
+		return undefined
+	}
+}
+
+export const updateWindowIfExists = async (tabId: number, updateProperties: browser.windows._UpdateUpdateInfo) => {
+	try {
+		const window = await browser.windows.update(tabId, updateProperties)
+		checkAndThrowRuntimeLastError()
+		return window
+	} catch (e: unknown){
+		return undefined
+	}
+}
+
+export const checkAndThrowRuntimeLastError = () => {
```

Comment:
not really an issue but was wondering about the decision to throw error here as opposed to safe parsing?

Reply:
what do you mean by safe parsing? What is should be parsed into?